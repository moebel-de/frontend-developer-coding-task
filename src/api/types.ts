interface Temp {
  day: number;
  min: number;
  max: number;
  night: number;
  eve: number;
  morn: number;
}

interface FeelsLike {
  day: number;
  night: number;
  eve: number;
  morn: number;
}

interface Weather {
  id: number;
  main: string;
  description: string;
  icon: string;
}

interface Day {
  dt: number;
  sunrise: number;
  sunset: number;
  pressure: number;
  humidity: number;
  dew_point: number;
  uvi: number;
  clouds: number;
  wind_speed: number;
  wind_deg: number;
  weather: Weather[];
}

export interface CurrentDay extends Day {
  temp: number;
  visibility: number;
  feels_like: number;
}

export interface Daily extends Day {
  feels_like: FeelsLike;
  pop: number;
  rain?: number;
  temp: Temp;
}

export interface WeeklyForecast {
  lat: number;
  lon: number;
  timezone: string;
  timezone_offset: number;
  current: CurrentDay;
  daily: Daily[];
}
