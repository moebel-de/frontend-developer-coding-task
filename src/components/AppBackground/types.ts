import { palette } from "./palette";

export type BackgroundType = "cold" | "hot";

export enum BackgroundTypes {
  cold = "cold",
  hot = "hot",
}
export interface Gradient {
  top: palette;
  bottom: palette;
}

export type Gradients = {
  [key in BackgroundType]: Gradient;
};
