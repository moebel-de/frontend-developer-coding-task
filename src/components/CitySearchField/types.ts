import { ChangeEvent, FocusEvent } from "react";

type FocusBlurHandler = (
  event: FocusEvent<HTMLInputElement> | FocusEvent<HTMLButtonElement>
) => void;

export interface CitySearchFieldProps {
  value: string;
  onSearch: () => void;
  onBlur?: FocusBlurHandler;
  onFocus?: FocusBlurHandler;
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
}
