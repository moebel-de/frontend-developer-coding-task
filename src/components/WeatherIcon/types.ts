import { CSSProperties, AriaAttributes } from "react";

interface HTMLProps extends AriaAttributes {
  className?: string;
}

export interface WeatherIconProps extends HTMLProps {
  color?: string;
  description: string;
  id: string;
  iconId: number;
  size?: IconSize;
  style?: CSSProperties;
  title: string;
}

export enum IconSize {
  small = "small",
  medium = "medium",
  large = "large",
  xlarge = "xlarge",
  xxlarge = "xxlarge",
}
