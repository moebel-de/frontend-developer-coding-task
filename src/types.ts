export interface WeatherData {
  dt: number;
  sunrise: number;
  sunset: number;
  temp: {
    day: number;
    min: number;
    max: number;
    night: number;
    eve: number;
    morn: number;
  };
  feels_like: {
    day: number;
    night: number;
    eve: number;
    morn: number;
  };
  pressure: number;
  humidity: number;
  weather: [
    {
      id: number;
      main: string;
      description: string;
      icon: string;
    },
  ];
  speed: number;
  deg: number;
  clouds: number;
  pop: number;
  rain?: number;
}

export type WeatherDataForCity = {
  city: {
    name: string;
    id: number;
    coord: {
      lon: number;
      lat: number;
    };
    country: string;
    population: number;
    timezone: number;
  };
  list: WeatherData[];
  cod: string;
  message: number;
  cnt: number;
};
