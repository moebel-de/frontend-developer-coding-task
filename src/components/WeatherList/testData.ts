import { WeatherDataForCity } from "../../types";

export const testData: WeatherDataForCity = {
  city: {
    id: 2643743,
    name: "London",
    coord: {
      lon: -0.1257,
      lat: 51.5085,
    },
    country: "GB",
    population: 1000000,
    timezone: 3600,
  },
  cod: "200",
  message: 0.0578669,
  cnt: 16,
  list: [
    {
      dt: 1595678400,
      sunrise: 1595650463,
      sunset: 1595707162,
      temp: {
        day: 18.18,
        min: 16.03,
        max: 18.45,
        night: 16.03,
        eve: 18.45,
        morn: 18.18,
      },
      feels_like: {
        day: 15.31,
        night: 12.99,
        eve: 15.37,
        morn: 15.31,
      },
      pressure: 1007,
      humidity: 88,
      weather: [
        {
          id: 501,
          main: "Rain",
          description: "moderate rain",
          icon: "10d",
        },
      ],
      speed: 7.02,
      deg: 218,
      clouds: 100,
      pop: 1,
      rain: 6.04,
    },
    {
      dt: 1595764800,
      sunrise: 1595736949,
      sunset: 1595793477,
      temp: {
        day: 20.97,
        min: 15.05,
        max: 23.04,
        night: 16.79,
        eve: 20.6,
        morn: 15.05,
      },
      feels_like: {
        day: 17.02,
        night: 14.34,
        eve: 17.37,
        morn: 11.75,
      },
      pressure: 1011,
      humidity: 50,
      weather: [
        {
          id: 500,
          main: "Rain",
          description: "light rain",
          icon: "10d",
        },
      ],
      speed: 5.76,
      deg: 253,
      clouds: 67,
      pop: 0.47,
      rain: 1.33,
    },
    {
      dt: 1595851200,
      sunrise: 1595823436,
      sunset: 1595879790,
      temp: {
        day: 17.69,
        min: 16.62,
        max: 18.72,
        night: 16.87,
        eve: 18.54,
        morn: 16.62,
      },
      feels_like: {
        day: 13.73,
        night: 15.69,
        eve: 15.74,
        morn: 14.49,
      },
      pressure: 1009,
      humidity: 68,
      weather: [
        {
          id: 500,
          main: "Rain",
          description: "light rain",
          icon: "10d",
        },
      ],
      speed: 6.42,
      deg: 194,
      clouds: 92,
      pop: 0.81,
      rain: 1.35,
    },
    {
      dt: 1595937600,
      sunrise: 1595909924,
      sunset: 1595966101,
      temp: {
        day: 20.52,
        min: 14.97,
        max: 20.52,
        night: 14.97,
        eve: 19.09,
        morn: 15.54,
      },
      feels_like: {
        day: 16.68,
        night: 11.96,
        eve: 15.01,
        morn: 13.59,
      },
      pressure: 1013,
      humidity: 43,
      weather: [
        {
          id: 802,
          main: "Clouds",
          description: "scattered clouds",
          icon: "03d",
        },
      ],
      speed: 4.65,
      deg: 284,
      clouds: 44,
      pop: 0,
    },
    {
      dt: 1596024000,
      sunrise: 1595996412,
      sunset: 1596052411,
      temp: {
        day: 20.52,
        min: 13.59,
        max: 21.07,
        night: 16.91,
        eve: 20.38,
        morn: 13.59,
      },
      feels_like: {
        day: 16.86,
        night: 15.67,
        eve: 17.91,
        morn: 11.04,
      },
      pressure: 1020,
      humidity: 39,
      weather: [
        {
          id: 803,
          main: "Clouds",
          description: "broken clouds",
          icon: "04d",
        },
      ],
      speed: 3.94,
      deg: 274,
      clouds: 57,
      pop: 0,
    },
    {
      dt: 1596110400,
      sunrise: 1596082902,
      sunset: 1596138718,
      temp: {
        day: 20.93,
        min: 16.23,
        max: 23.8,
        night: 18.68,
        eve: 23.33,
        morn: 16.23,
      },
      feels_like: {
        day: 20.23,
        night: 16.86,
        eve: 21.45,
        morn: 14.99,
      },
      pressure: 1021,
      humidity: 48,
      weather: [
        {
          id: 804,
          main: "Clouds",
          description: "overcast clouds",
          icon: "04d",
        },
      ],
      speed: 0.88,
      deg: 275,
      clouds: 97,
      pop: 0,
    },
    {
      dt: 1596196800,
      sunrise: 1596169392,
      sunset: 1596225023,
      temp: {
        day: 24.69,
        min: 17.63,
        max: 24.69,
        night: 17.63,
        eve: 21.97,
        morn: 17.81,
      },
      feels_like: {
        day: 21.09,
        night: 14.97,
        eve: 18.34,
        morn: 15.84,
      },
      pressure: 1018,
      humidity: 37,
      weather: [
        {
          id: 802,
          main: "Clouds",
          description: "scattered clouds",
          icon: "03d",
        },
      ],
      speed: 4.84,
      deg: 91,
      clouds: 43,
      pop: 0,
    },
    {
      dt: 1596283200,
      sunrise: 1596255883,
      sunset: 1596311327,
      temp: {
        day: 25.19,
        min: 18.15,
        max: 25.82,
        night: 18.54,
        eve: 23.72,
        morn: 18.15,
      },
      feels_like: {
        day: 21.6,
        night: 17.24,
        eve: 20.99,
        morn: 16.68,
      },
      pressure: 1013,
      humidity: 36,
      weather: [
        {
          id: 800,
          main: "Clear",
          description: "sky is clear",
          icon: "01d",
        },
      ],
      speed: 4.83,
      deg: 90,
      clouds: 4,
      pop: 0.02,
    },
    {
      dt: 1596369600,
      sunrise: 1596342374,
      sunset: 1596397628,
      temp: {
        day: 25.19,
        min: 18.26,
        max: 25.19,
        night: 20.06,
        eve: 24.21,
        morn: 18.26,
      },
      feels_like: {
        day: 23.55,
        night: 19.11,
        eve: 22.81,
        morn: 17.43,
      },
      pressure: 1011,
      humidity: 47,
      weather: [
        {
          id: 804,
          main: "Clouds",
          description: "overcast clouds",
          icon: "04d",
        },
      ],
      speed: 3.7,
      deg: 59,
      clouds: 98,
      pop: 0.16,
    },
    {
      dt: 1596456000,
      sunrise: 1596428866,
      sunset: 1596483928,
      temp: {
        day: 27.06,
        min: 17.59,
        max: 28.68,
        night: 21.42,
        eve: 26.91,
        morn: 17.59,
      },
      feels_like: {
        day: 26.32,
        night: 22.35,
        eve: 26.2,
        morn: 16.74,
      },
      pressure: 1013,
      humidity: 37,
      weather: [
        {
          id: 800,
          main: "Clear",
          description: "sky is clear",
          icon: "01d",
        },
      ],
      speed: 1.57,
      deg: 18,
      clouds: 0,
      pop: 0,
    },
    {
      dt: 1596542400,
      sunrise: 1596515359,
      sunset: 1596570226,
      temp: {
        day: 24.97,
        min: 18.32,
        max: 24.97,
        night: 18.32,
        eve: 20.14,
        morn: 20.5,
      },
      feels_like: {
        day: 24.58,
        night: 16.25,
        eve: 20.76,
        morn: 20.19,
      },
      pressure: 1012,
      humidity: 52,
      weather: [
        {
          id: 502,
          main: "Rain",
          description: "heavy intensity rain",
          icon: "10d",
        },
      ],
      speed: 2.57,
      deg: 269,
      clouds: 99,
      pop: 1,
      rain: 19.15,
    },
    {
      dt: 1596628800,
      sunrise: 1596601852,
      sunset: 1596656523,
      temp: {
        day: 21.74,
        min: 15.33,
        max: 23.31,
        night: 17.82,
        eve: 21.8,
        morn: 15.33,
      },
      feels_like: {
        day: 18.76,
        night: 15.84,
        eve: 19.32,
        morn: 11.77,
      },
      pressure: 1014,
      humidity: 46,
      weather: [
        {
          id: 802,
          main: "Clouds",
          description: "scattered clouds",
          icon: "03d",
        },
      ],
      speed: 4.17,
      deg: 324,
      clouds: 36,
      pop: 0.55,
    },
    {
      dt: 1596715200,
      sunrise: 1596688345,
      sunset: 1596742818,
      temp: {
        day: 23.4,
        min: 16.74,
        max: 24.53,
        night: 19.2,
        eve: 24.42,
        morn: 16.74,
      },
      feels_like: {
        day: 21.64,
        night: 18.22,
        eve: 22.67,
        morn: 15.25,
      },
      pressure: 1018,
      humidity: 50,
      weather: [
        {
          id: 803,
          main: "Clouds",
          description: "broken clouds",
          icon: "04d",
        },
      ],
      speed: 3.57,
      deg: 274,
      clouds: 81,
      pop: 0,
    },
    {
      dt: 1596801600,
      sunrise: 1596774839,
      sunset: 1596829111,
      temp: {
        day: 25.87,
        min: 17.08,
        max: 26.18,
        night: 19.42,
        eve: 23.86,
        morn: 17.08,
      },
      feels_like: {
        day: 24.02,
        night: 18.31,
        eve: 22.57,
        morn: 15.62,
      },
      pressure: 1018,
      humidity: 48,
      weather: [
        {
          id: 802,
          main: "Clouds",
          description: "scattered clouds",
          icon: "03d",
        },
      ],
      speed: 4.46,
      deg: 257,
      clouds: 46,
      pop: 0,
    },
    {
      dt: 1596888000,
      sunrise: 1596861333,
      sunset: 1596915403,
      temp: {
        day: 27.38,
        min: 19.38,
        max: 28.39,
        night: 22.26,
        eve: 26.99,
        morn: 19.38,
      },
      feels_like: {
        day: 27.35,
        night: 23.24,
        eve: 26.85,
        morn: 18.8,
      },
      pressure: 1020,
      humidity: 53,
      weather: [
        {
          id: 802,
          main: "Clouds",
          description: "scattered clouds",
          icon: "03d",
        },
      ],
      speed: 3.41,
      deg: 279,
      clouds: 35,
      pop: 0,
    },
    {
      dt: 1596974400,
      sunrise: 1596947828,
      sunset: 1597001693,
      temp: {
        day: 28.57,
        min: 20.33,
        max: 30.42,
        night: 24.41,
        eve: 29.18,
        morn: 20.33,
      },
      feels_like: {
        day: 28.26,
        night: 25.44,
        eve: 28.75,
        morn: 21.2,
      },
      pressure: 1022,
      humidity: 40,
      weather: [
        {
          id: 800,
          main: "Clear",
          description: "sky is clear",
          icon: "01d",
        },
      ],
      speed: 2.07,
      deg: 283,
      clouds: 0,
      pop: 0,
    },
  ],
};
