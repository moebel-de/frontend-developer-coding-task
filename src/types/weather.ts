export interface WeekDayWeather {
  id: string;
  date: number;
  temperature: number;
  iconId: number;
  name: string;
  description: string;
}
