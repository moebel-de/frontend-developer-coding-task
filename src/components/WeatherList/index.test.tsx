import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { WeatherList } from ".";
import "jest-styled-components";
import { testData as data } from "./testData";
import { WeatherData } from "../../types";
const testData = data.list;

describe("WeatherList", () => {
  test("test WeatherList without blur", () => {
    const { container } = render(<WeatherList data={testData} />);
    expect(container).toMatchSnapshot();
    expect(container).toBeInTheDocument();
    expect(container.firstChild).not.toHaveStyleRule("filter", "blur(8px)");
  });
  test("test WeatherList loading", () => {
    const { getByText } = render(<WeatherList loading />);

    expect(getByText("Loading...")).toBeInTheDocument();
  });
  test("test WeatherList with blur", () => {
    const { container } = render(<WeatherList data={testData} isBlurred />);
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule("filter", "blur(8px)");
  });
  test("test WeatherList empty data", () => {
    const data: WeatherData[] = [];
    const { container } = render(<WeatherList data={data} isBlurred />);

    expect(container.firstChild).toBeNull();
  });
  test("test WeatherList empty data", () => {
    const data: WeatherData[] = [];
    const { container } = render(<WeatherList data={data} isBlurred />);

    expect(container.firstChild).toBeNull();
  });
  test("test WeatherList empty data", () => {
    const data: WeatherData[] = [];

    const errorMessage = "WhateverErrorText";
    const { container } = render(
      <WeatherList errorMessage={errorMessage} data={data} isBlurred />,
    );
    expect(screen.getByText(errorMessage)).toBeInTheDocument();
    expect(container.firstChild).toBeInTheDocument();
  });
  test("test WeatherList click more", () => {
    const {
      container: { firstChild },
    } = render(<WeatherList data={testData} isBlurred />);
    expect(firstChild?.childNodes.length).toBe(6);

    expect(screen.getByText(/more/)).toBeDefined();
    fireEvent.click(firstChild?.lastChild!!);
    expect(firstChild?.childNodes.length).toBe(17);
    expect(screen.getByText(/less/)).toBeDefined();
  });
});
