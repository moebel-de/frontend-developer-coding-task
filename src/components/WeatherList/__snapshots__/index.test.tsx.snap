// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherList test WeatherList with blur 1`] = `
.c2 {
  width: 60px;
}

.c0 {
  width: 100%;
  max-width: 500px;
  margin: 0 auto;
  margin-top: 40px;
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  -webkit-filter: blur(8px);
  filter: blur(8px);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 28px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 60px;
}

.c3 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 div:nth-child(1) {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c3 div:nth-child(2) {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 28px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 60px;
  background-color: #ffffff55;
  border-radius: 12px;
  padding: 0 10px;
  cursor: pointer;
  margin: 0 -10px;
}

.c4 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 div:nth-child(1) {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c4 div:nth-child(2) {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 80px;
  font-weight: 900;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 > * {
  padding: 0 10px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
      >
        <svg>
          wi-day-showers.svg
        </svg>
      </span>
      <div>
        18
        °
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        Sunday
      </div>
      <div>
        21
        °
      </div>
      <div>
        <span
          class="c2"
        >
          <svg>
            wi-day-showers.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        Monday
      </div>
      <div>
        18
        °
      </div>
      <div>
        <span
          class="c2"
        >
          <svg>
            wi-day-showers.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        Tuesday
      </div>
      <div>
        21
        °
      </div>
      <div>
        <span
          class="c2"
        >
          <svg>
            wi-cloudy.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        Wednesday
      </div>
      <div>
        21
        °
      </div>
      <div>
        <span
          class="c2"
        >
          <svg>
            wi-cloudy.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="c4"
    >
      <div>
        show 
        more
      </div>
      <div />
      <div>
        <span
          class="c2"
        >
          <svg>
            wi-direction-down.svg
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`WeatherList test WeatherList without blur 1`] = `
.c2 {
  width: 60px;
}

.c0 {
  width: 100%;
  max-width: 500px;
  margin: 0 auto;
  margin-top: 40px;
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 28px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 60px;
}

.c3 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 div:nth-child(1) {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c3 div:nth-child(2) {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 28px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 60px;
  background-color: #ffffff55;
  border-radius: 12px;
  padding: 0 10px;
  cursor: pointer;
  margin: 0 -10px;
}

.c4 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 div:nth-child(1) {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c4 div:nth-child(2) {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 80px;
  font-weight: 900;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 > * {
  padding: 0 10px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
      >
        <svg>
          wi-day-showers.svg
        </svg>
      </span>
      <div>
        18
        °
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        Sunday
      </div>
      <div>
        21
        °
      </div>
      <div>
        <span
          class="c2"
        >
          <svg>
            wi-day-showers.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        Monday
      </div>
      <div>
        18
        °
      </div>
      <div>
        <span
          class="c2"
        >
          <svg>
            wi-day-showers.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        Tuesday
      </div>
      <div>
        21
        °
      </div>
      <div>
        <span
          class="c2"
        >
          <svg>
            wi-cloudy.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        Wednesday
      </div>
      <div>
        21
        °
      </div>
      <div>
        <span
          class="c2"
        >
          <svg>
            wi-cloudy.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="c4"
    >
      <div>
        show 
        more
      </div>
      <div />
      <div>
        <span
          class="c2"
        >
          <svg>
            wi-direction-down.svg
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
